### PR TITLE
Add support for analyzers from ref-packs

### PR DESCRIPTION
**Pull request by ericstj**
Also add analyzers to conflict resolution.

Implements https://github.com/dotnet/designs/blob/main/accepted/2021/InboxSourceGenerators.md

### CODE REVIEW

**Comment by ericstj** [comment]
@dsplaisted @wli3 can one of you have a look at this and give some guidance around what might be missing?  I did what I think might be the minimum to enable the scenario but I expect to need to do more :)

File: src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs [hunk 1]
```diff
@@ -234,17 +243,40 @@ private void AddNetStandardTargetingPackAssets(ITaskItem targetingPack, string t
                     continue;
                 }
 
-                string dllPath = usePathElementsInFrameworkListAsFallBack ?
+                string itemType = fileElement.Attribute("Type")?.Value;
+                bool isAnalzer = itemType?.Equals("Analyzer", StringComparison.OrdinalIgnoreCase) ?? false;
```
Comment (dsplaisted) [inline src/Tasks/Microsoft.NET.Build.Tasks/ResolveTargetingPackAssets.cs:247]: Typo: `isAnalzer` -> `isAnalyzer`

**Review by dsplaisted — APPROVED** [review]
Looks good.  Is there any way to have test coverage of analyzer conflict resolution?

**Comment by ericstj** [comment]
Did you have a suggestion for where to look for such tests, @dsplaisted?  I saw tests of the conflict resolver itself, but nothing that tested the task level where this logic lives.  I might be missing something though.

**Comment by ericstj** [comment]
Perhaps here: https://github.com/dotnet/sdk/blob/main/src/Tests/Microsoft.NET.Build.Tests/GivenThatWeWantToResolveConflicts.cs?

**Review by safern — APPROVED** [review]
Looks good.

**Comment by ericstj** [comment]
I added a test that forced inbox and OOB SDK analyzers to be used.  I made it fail locally by disabling the conflict-resolution change in targets:
```
C:\src\dotnet\sdk\artifacts\bin\redist\Debug\dotnet\sdk\6.0.100-dev\Sdks\Microsoft.NET.Sdk\analyzers\build\Microsoft.CodeAnalysis.NetAnalyzers.targets(580,5): warning : The .NET SDK has newer analyzers with version '6.0.0' than what version '5.0.3' of 'Microsoft.CodeAnalysis.NetAnalyzers' package provides. Update or remove this package reference. [C:\src\dotnet\sdk\artifacts\tmp\Debug\AnalyzersAreC---4DE360D4\AnalyzersAreConflictResolved\AnalyzersAreConflictResolved.csproj]
CSC : error CS8034: Unable to load Analyzer assembly C:\src\dotnet\sdk\artifacts\.nuget\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\cs\Microsoft.CodeAnalysis.CSharp.NetAnalyzers.dll : Assembly with same name is already loaded [C:\src\dotnet\sdk\artifacts\tmp\Debug\AnalyzersAreC---4DE360D4\AnalyzersAreConflictResolved\AnalyzersAreConflictResolved.csproj]
CSC : error CS8034: Unable to load Analyzer assembly C:\src\dotnet\sdk\artifacts\.nuget\packages\microsoft.codeanalysis.netanalyzers\5.0.3\analyzers\dotnet\cs\Microsoft.CodeAnalysis.NetAnalyzers.dll : Assembly with same name is already loaded [C:\src\dotnet\sdk\artifacts\tmp\Debug\AnalyzersAreC---4DE360D4\AnalyzersAreConflictResolved\AnalyzersAreConflictResolved.csproj]
```
Then put my change back in place and it succeeds.

I think this is ready to go.

**Review by dsplaisted — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets [hunk 2]
```diff
@@ -326,12 +326,14 @@ Copyright (c) .NET Foundation. All rights reserved.
...
 
     <ResolveTargetingPackAssets FrameworkReferences="@(FrameworkReference)"
+                                ProjectLanguage="$(Language)"
                                 ResolvedTargetingPacks="@(ResolvedTargetingPack)"
                                 RuntimeFrameworks="@(RuntimeFramework)"
                                 GenerateErrorForMissingTargetingPacks="$(GenerateErrorForMissingTargetingPacks)"
                                 NuGetRestoreSupported="$(_NuGetRestoreSupported)"
                                 NetCoreTargetingPackRoot="$(NetCoreTargetingPackRoot)">
       <Output TaskParameter="ReferencesToAdd" ItemName="Reference" />
+      <Output TaskParameter="AnalyzersToAdd" Condition="'$(DisableFrameworkReferenceAnalyzers)' != 'true'" ItemName="Analyzer" />
```
Comment (ViktorHofer) [inline src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets:336]: Should the `DisableFrameworkReferenceAnalyzers` property be documented somewhere?

**Review by ViktorHofer — APPROVED** [review]

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets (hunk 2 above)
Comment (ericstj) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets:336]: At the moment it's written here: https://github.com/dotnet/designs/blob/main/accepted/2021/InboxSourceGenerators.md

I think official docs should be added here:
https://github.com/dotnet/docs/blob/211a979fab01259e7c0ce8d1cd737b1ea4ba26f5/docs/core/project-sdk/msbuild-props.md

**Comment by ericstj** [comment]
Tested this with the work I have in PR in arcade/runtime and it's working as expected.  Thanks folks!

File: src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets (hunk 2 above)
Comment (ericstj) [reply src/Tasks/Microsoft.NET.Build.Tasks/targets/Microsoft.NET.Sdk.FrameworkReferenceResolution.targets:336]: https://github.com/dotnet/docs/pull/24896

**Pull request merged**